includes/API/FBE/Configuration/Update/Request.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package FacebookCommerce
+ */
+
+namespace SkyVerge\WooCommerce\Facebook\API\FBE\Configuration\Update;
+
+defined( 'ABSPATH' ) or exit;
+
+use SkyVerge\WooCommerce\Facebook\API\FBE\Configuration;
+
+/**
+ * FBE Configuration update request object.
+ *
+ * @since 2.0.0-dev.1
+ */
+class Request extends Configuration\Request  {
+
+
+	/**
+	 * API request constructor.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param string $external_business_id external business ID
+	 */
+	public function __construct( $external_business_id ) {
+
+		parent::__construct( $external_business_id, 'POST' );
+
+		// include the business ID in the request body
+		$this->data['fbe_external_business_id'] = $external_business_id;
+	}
+
+
+	/**
+	 * Sets the messenger configuration.
+	 *
+	 * Only the enabled and domains values are able to accept updates right now.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param Configuration\Messenger $configuration messenger configuration object
+	 */
+	public function set_messenger_configuration( Configuration\Messenger $configuration ) {
+
+		$this->data['messenger_chat'] = [
```

Comment:
Maybe we need to merge our values with the original values from the API here, as not to reset the color/greeting.

Reply:
Or maybe only send the `enabled` and `domains` keys, not the full array.

Reply:
I noticed this too. If you follow the steps below, the selected greeting and color are removed from the settings on Facebook.

1. Uncheck Enable Messenger
1. Save changes (at this point you can click Add again for Messenger plugin on Facebook and the settings are still there)
1. Check Enable Messenger
1. Save changes (the Messenger plugin is added again, but the color and greeting settings are reset)